digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/LaneUtils.scala

```diff
@@ -0,0 +1,253 @@
+
+      if (road.roadPart > laneRoadAddressInfo.initialRoadPartNumber && road.roadPart < laneRoadAddressInfo.endRoadPartNumber) {
+        ( road.startMValue, road.endMValue)
+
+      }
+      else if (road.roadPart == laneRoadAddressInfo.initialRoadPartNumber && road.roadPart == laneRoadAddressInfo.endRoadPartNumber) {
+
+        if (!(road.endAddrMValue > laneRoadAddressInfo.initialDistance && road.startAddrMValue < laneRoadAddressInfo.endDistance))
+          (None, None)
+
+        else if (road.startAddrMValue <= laneRoadAddressInfo.initialDistance && road.endAddrMValue >= laneRoadAddressInfo.endDistance) {
+          ( startPoint, endPoint )
+
+        }
+        else if (road.startAddrMValue <= laneRoadAddressInfo.initialDistance && road.endAddrMValue < laneRoadAddressInfo.endDistance) {
+          ( startPoint, road.endMValue )
+
+        }
+        else if (road.startAddrMValue > laneRoadAddressInfo.initialDistance && road.endAddrMValue >= laneRoadAddressInfo.endDistance) {
+          ( road.startMValue, endPoint )
+
+        }
+        else {
+          ( road.startMValue, road.endMValue )
+
+        }
+
+      }
+      else if (road.roadPart == laneRoadAddressInfo.initialRoadPartNumber) {
+        if (road.endAddrMValue < laneRoadAddressInfo.initialDistance) {
+          (None, None)
+
+        } else if (road.startAddrMValue <= laneRoadAddressInfo.initialDistance) {
+          ( startPoint, road.endMValue )
+
+        } else {
+          ( road.startMValue, road.endMValue )
+        }
+
+      }
+      else if (road.roadPart == laneRoadAddressInfo.endRoadPartNumber) {
+        if (road.endAddrMValue < laneRoadAddressInfo.endDistance) {
+          (None, None)
+
+        } else if (road.endAddrMValue >= laneRoadAddressInfo.endDistance) {
+          ( road.startMValue, endPoint )
+
+        } else {
+          ( road.startMValue, road.endMValue )
+
+        }
+      }
+      else {
+        (None, None)
+      }
+    }
+
+    // Main process
+    def process() = {
+
+      val filteredRoadAddresses = getRoadAddressToProcess()
+
+      //Get only the lanes to create
+      val lanesToInsert = newIncomeLanes.filter(_.id == 0)
+
+
+      val allLanesToCreate = filteredRoadAddresses.flatMap { road =>
+        val vvhTimeStamp = vvhClient.roadLinkData.createVVHTimeStamp()
+
+        lanesToInsert.flatMap { lane =>
+          val laneCodeProperty = lane.properties.find(_.publicId == "lane_code")
+                                                .getOrElse(throw new IllegalArgumentException("Lane Code attribute not found!"))
+
+          val laneCodeValue = laneCodeProperty.values.head.value.toString
+          val laneCode = if ( laneCodeValue.nonEmpty )
+                           laneCodeValue.toInt
+                         else
+                           throw new IllegalArgumentException("Lane Code attribute Empty!")
+
+          val isMainLane = laneCodeValue.charAt(1).getNumericValue == 1
```

Comment:
Apply new code values for code lanes to validate if it's the main lane

Reply:
Done